test/unit/test_telemetry.py

```diff
@@ -235,3 +242,99 @@ def test_generate_telemetry_data():
         }
         and telemetry_data.timestamp == 123
     )
+
+
+def test_raising_error_generates_telemetry_event_when_connection_is_present():
+    mock_connection = get_mocked_telemetry_connection()
+
+    with pytest.raises(RevocationCheckError):
+        raise RevocationCheckError(
+            msg="Response unavailable",
+            errno=ER_OCSP_RESPONSE_UNAVAILABLE,
+            connection=mock_connection,
+            send_telemetry=True,
+        )
+
+    mock_connection._log_telemetry.assert_called_once()
+    verify_telemetry_data(mock_connection._log_telemetry.call_args[0][0])
+
+
+def test_raising_error_with_send_telemetry_off_does_not_generate_telemetry_event_when_connection_is_present():
+    mock_connection = get_mocked_telemetry_connection()
+
+    with pytest.raises(RevocationCheckError):
+        raise RevocationCheckError(
+            msg="Response unavailable",
+            errno=ER_OCSP_RESPONSE_UNAVAILABLE,
+            connection=mock_connection,
+            send_telemetry=False,
+        )
+
+    mock_connection._log_telemetry.assert_not_called()
+
+
+def test_request_throws_revocation_check_error():
+    retry_ctx = Mock()
+    retry_ctx.current_retry_count = 0
+    retry_ctx.timeout = 10
+    retry_ctx.add_retry_params.return_value = "https://example.com"
+
+    mock_connection = get_mocked_telemetry_connection()
+
+    with mock.patch.object(SnowflakeRestful, "_request_exec") as _request_exec_mocked:
+        _request_exec_mocked.side_effect = RevocationCheckError(
+            msg="Response unavailable", errno=ER_OCSP_RESPONSE_UNAVAILABLE
+        )
+        mock_restful = SnowflakeRestful(connection=mock_connection)
+        with pytest.raises(RevocationCheckError):
+            mock_restful._request_exec_wrapper(
+                None,
+                None,
+                None,
+                None,
+                None,
+                retry_ctx,
+            )
+        mock_restful._connection._log_telemetry.assert_called_once()
+        verify_telemetry_data(mock_connection._log_telemetry.call_args[0][0])
+
+
+def get_mocked_telemetry_connection(telemetry_enabled: bool = True) -> Mock:
+    mock_connection = Mock()
+    mock_connection.application = "test_application"
+    mock_connection.telemetry_enabled = telemetry_enabled
+    mock_connection.is_closed = False
+
+    mock_connection._log_telemetry = Mock()
+
+    mock_telemetry = Mock()
+    mock_telemetry.is_closed = False
+    mock_connection._telemetry = mock_telemetry
+
+    return mock_connection
+
+
+def verify_telemetry_data(telemetry_data: TelemetryData):
```

Comment:
I'd rename that as this checks particularly RevocationCheckError (or use dict comparison like in other tests)

```suggestion
def assert_telemetry_data_for_revocation_check_error(telemetry_data: TelemetryData):
```

Reply:
Done